provisioning/device/src/interfaces.ts

```diff
@@ -6,61 +6,50 @@
 import { SharedAccessSignature, X509 } from 'azure-iot-common';
 
 /**
- * @private
- * Callback for responses from the previsioning service
- */
-export type ResponseCallback = (err?: Error, response?: any) => void;
-
-/**
  * type defining the types of authentication we support
  */
-export type Authentication = string | X509 | SharedAccessSignature;
-
-export interface ClientConfiguration {
-  /**
-   * User-Agent string passed to the service as part of communication
-   */
-  userAgent: string;
+export type ProvisioningAuthentication = string | X509 | SharedAccessSignature;
```

Comment:
I think that'd be TPM. it ends up using SAS tokens but the flow is typical of TPM (and yes, it's asymmetrical with x509 but I think that's what other SDKs landed on)